### PR TITLE
preload chats

### PR DESCRIPTION
**Pull request by flexsurfer**
we want to first preload N chats and then load all

### CODE REVIEW

File: protocol/messenger_chats.go [hunk 1]
```diff
@@ -9,12 +9,13 @@ import (
...
-func (m *Messenger) Chats() []*Chat {
+func (m *Messenger) Chats(num int) []*Chat {
 	var chats []*Chat
-
+	var i = 0
 	m.allChats.Range(func(chatID string, chat *Chat) (shouldContinue bool) {
 		chats = append(chats, chat)
-		return true
+		i++
+		return num == 0 || i < num
```
Comment (Samyoul) [inline protocol/messenger_chats.go]: I can see what you are doing here, but I think it is introducing a bit of an anti-pattern. Giving a zero parameter for all chats is a bit wonky. In my mind there are 2 other options.

a change to the function to allow for no parameter, using a variadic parameter. 

```go
func (m *Messenger) Chats(num ...int) []*Chat {
	var chats []*Chat
	var i int
	
	m.allChats.Range(func(chatID string, chat *Chat) (shouldContinue bool) {
		chats = append(chats, chat)
		i++
		return len(num) == 0 || i < num[0]
	})

	return chats
}
```

Can be called 

`s.m.Chats()` or `s.m.Chats(4)`

OR

A dedicated function that is designed specifically to return `n` Chats. And a parameter of 0 will return 0 Chats.

Comment (Samyoul) [reply protocol/messenger_chats.go]: Also would it be possible to add some tests for the new behaviour of this function or new dedicated function?

Comment (flexsurfer) [reply protocol/messenger_chats.go]: good points ,thanks

**Comment by flexsurfer** [comment]
@Samyoul done

**Review by Samyoul — APPROVED** [review]
Looks good 👍

**Pull request merged**